src/routes/transactions/mappers/common/transaction-data.mapper.spec.ts

```diff
@@ -0,0 +1,525 @@
+import { faker } from '@faker-js/faker';
+import { ContractsRepository } from '../../../../domain/contracts/contracts.repository';
+import { contractBuilder } from '../../../../domain/contracts/entities/__tests__/contract.builder';
+import {
+  dataDecodedBuilder,
+  dataDecodedParameterBuilder,
+} from '../../../../domain/data-decoder/entities/__tests__/data-decoded.builder';
+import { AddressInfoHelper } from '../../../common/address-info/address-info.helper';
+import { NULL_ADDRESS } from '../../../common/constants';
+import { AddressInfo } from '../../../common/entities/address-info.entity';
+import { MULTI_SEND_METHOD_NAME } from '../../constants';
+import { DataDecodedParamHelper } from './data-decoded-param.helper';
+import { TransactionDataMapper } from './transaction-data.mapper';
+
+const addressInfoHelper = jest.mocked({
+  getOrDefault: jest.fn(),
+} as unknown as AddressInfoHelper);
+
+const contractsRepository = jest.mocked({
+  getContract: jest.fn(),
+} as unknown as ContractsRepository);
+
+const dataDecodedParamHelper = jest.mocked({
+  hasNestedDelegate: jest.fn(),
+} as unknown as DataDecodedParamHelper);
+
+describe('Transaction Data Mapper (Unit)', () => {
+  const mapper = new TransactionDataMapper(
```

Comment:
We should move this to `beforeEach` so that we can avoid potential state sharing between the tests in the future

Reply:
Yes, I agree. In this case, the mapper instance is built just using mocks, so they are reset for each test, but it's a best practice and also covers the case the mapper actually contains state in the future 👍🏻 I've changed it in [9565c0a](https://github.com/5afe/safe-client-gateway-nest/pull/291/commits/9565c0ad8b6c6533745567f36f005f4915859bf4)

This could be applied to other parts of the code, I'm going to address those in a separate PR.

Reply:
Let's do it! 👍 

This is just to avoid breaking the tests if any internal state is introduced. It's can be trickier to debug those cases then.

Reply:
https://github.com/5afe/safe-client-gateway-nest/pull/301 🙂